util/config/config.go

```diff
@@ -356,175 +353,57 @@ func (config *Configuration) SetupPortMapping() error {
-
-		transport, err := transport.NewTransport(config.Transport)
-		if err != nil {
-			return err
-		}
+	var err error
+	gateway, err = portmapper.Discover()
+	if err == portmapper.NoGatewayFound {
+		log.Printf("No NAT gateway discovered, skip automatic port forwading. You need to set up port forwarding and firewall yourself.")
+		return nil
+	}
+	if err != nil {
+		return err
+	}
 
-		externalPort, internalPort, err := nat.AddPortMapping(transport.GetNetwork(), int(config.NodePort), int(config.NodePort), "NKN Node", config.NATPortMappingTimeout*time.Second)
-		if err != nil {
-			return err
-		}
-		log.Printf("Mapped external port %d to internal port %d", externalPort, internalPort)
+	err = gateway.Add(config.NodePort, "NKN Node")
+	if err != nil {
+		return err
+	}
+	log.Printf("Mapped external port %d to internal port %d", config.NodePort, config.NodePort)
 
-		externalPort, internalPort, err = nat.AddPortMapping(transport.GetNetwork(), int(config.HttpWsPort), int(config.HttpWsPort), "NKN Node", config.NATPortMappingTimeout*time.Second)
-		if err != nil {
-			return err
-		}
-		log.Printf("Mapped external port %d to internal port %d", externalPort, internalPort)
+	err = gateway.Add(config.HttpWsPort, "NKN Node")
+	if err != nil {
+		return err
+	}
+	log.Printf("Mapped external port %d to internal port %d", config.HttpWsPort, config.HttpWsPort)
 
-		externalPort, internalPort, err = nat.AddPortMapping(transport.GetNetwork(), int(config.HttpWssPort), int(config.HttpWssPort), "NKN Node", config.NATPortMappingTimeout*time.Second)
-		if err != nil {
-			return err
-		}
-		log.Printf("Mapped external port %d to internal port %d", externalPort, internalPort)
+	err = gateway.Add(config.HttpWssPort, "NKN Node")
+	if err != nil {
+		return err
+	}
+	log.Printf("Mapped external port %d to internal port %d", config.HttpWssPort, config.HttpWssPort)
 
-		externalPort, internalPort, err = nat.AddPortMapping(transport.GetNetwork(), int(config.HttpJsonPort), int(config.HttpJsonPort), "NKN Node", config.NATPortMappingTimeout*time.Second)
-		if err != nil {
-			return err
-		}
-		log.Printf("Mapped external port %d to internal port %d", externalPort, internalPort)
+	err = gateway.Add(config.HttpJsonPort, "NKN Node")
+	if err != nil {
+		return err
+	}
+	log.Printf("Mapped external port %d to internal port %d", config.HttpJsonPort, config.HttpJsonPort)
 
-		externalPort, internalPort, err = nat.AddPortMapping(transport.GetNetwork(), int(config.HttpsJsonPort), int(config.HttpsJsonPort), "NKN Node", config.NATPortMappingTimeout*time.Second)
-		if err != nil {
-			return err
-		}
-		log.Printf("Mapped external port %d to internal port %d", externalPort, internalPort)
+	err = gateway.Add(config.HttpsJsonPort, "NKN Node")
+	if err != nil {
+		return err
 	}
+	log.Printf("Mapped external port %d to internal port %d", config.HttpsJsonPort, config.HttpsJsonPort)
 
 	return nil
 }
 
 func (config *Configuration) ClearPortMapping() error {
-	if SkipNAT || !config.NAT {
```

Comment:
Why delete this condition judgment?

Reply:
Because it's the same as `gateway == nil` below